x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/mapper/SemanticTextFieldMapper.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ml.mapper;
+
+import org.apache.lucene.search.Query;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.index.fielddata.FieldDataContext;
+import org.elasticsearch.index.fielddata.IndexFieldData;
+import org.elasticsearch.index.mapper.DocumentParserContext;
+import org.elasticsearch.index.mapper.FieldMapper;
+import org.elasticsearch.index.mapper.MappedFieldType;
+import org.elasticsearch.index.mapper.MapperBuilderContext;
+import org.elasticsearch.index.mapper.SimpleMappedFieldType;
+import org.elasticsearch.index.mapper.SourceValueFetcher;
+import org.elasticsearch.index.mapper.TextSearchInfo;
+import org.elasticsearch.index.mapper.ValueFetcher;
+import org.elasticsearch.index.query.SearchExecutionContext;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ *  A {@link FieldMapper} for semantic text fields. These fields have a model id reference, that is used for performing inference
+ * at ingestion and query time.
+ * For now, it is compatible with text expansion models only, but will be extended to support dense vector models as well.
+ * This field mapper performs no indexing, as inference results will be included as a different field in the document source, and will
+ * be indexed using a different field mapper.
+ */
+public class SemanticTextFieldMapper extends FieldMapper {
+
+    public static final String CONTENT_TYPE = "semantic_text";
+
+    private static SemanticTextFieldMapper toType(FieldMapper in) {
+        return (SemanticTextFieldMapper) in;
+    }
+
+    private static Builder builder(FieldMapper in) {
+        return ((SemanticTextFieldMapper) in).builder;
+    }
+
+    public static class Builder extends FieldMapper.Builder {
+
+        private final Parameter<String> modelId = Parameter.stringParam("model_id", false, m -> builder(m).modelId.get(), null)
```

Comment:
```suggestion
        private final Parameter<String> modelId = Parameter.stringParam("model_id", false, m -> toType(m).fieldType().modelId, null)
```

Or at least don't create a builder here. That seems unnecessary.

Reply:
Good call - removing it